### PR TITLE
Product attribute fixtures improvements

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | -----
| Branch?         | 1.3
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes https://github.com/Sylius/Sylius/issues/9252
| License         | MIT

There was an issue with product attribute fixtures, that there was no possibility to use more sophisticated attributes in product fixtures (like `select` or `datetime`).  I decided to create a functional PHPUnit test for this bugfix, as Behat scenario would be a little bit strange and too technical, in my opinion.

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/Fixture/Factory/ProductExampleFactory.php [hunk 1]
```diff
@@ -339,6 +324,48 @@ private function getLocales(): iterable
...
+        $productAttribute = $this->productAttributeRepository->findOneBy(['code' => $code]);
+
+        Assert::notNull($productAttribute);
+
+        /** @var ProductAttributeValueInterface $productAttributeValue */
+        $productAttributeValue = $this->productAttributeValueFactory->createNew();
+        $productAttributeValue->setAttribute($productAttribute);
+
+        switch ($productAttribute->getType()) {
+            case DateAttributeType::TYPE:
```
Comment (lchrusciel) [inline src/Sylius/Bundle/CoreBundle/Fixture/Factory/ProductExampleFactory.php]: What would you say about this construction:
```php
        switch ($productAttribute->getType()) {
            case DateAttributeType::TYPE:
            case DatetimeAttributeType::TYPE:
                $productAttributeValue->setValue(new \DateTime($value));
                break;
            case CheckboxAttributeType::TYPE:
                $productAttributeValue->setValue($value);
                break;
            default:
                $productAttributeValue->setValue($value ?: $this->getRandomValueForProductAttribute($productAttribute));
        }

```

File: config/packages/test/fixtures.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# This file is part of the Sylius package.
+# (c) Paweł Jędrzejewski
+
+sylius_fixtures:
+    suites:
+        test:
```
Comment (lchrusciel) [inline config/packages/test/fixtures.yaml]: I'm not convinced about this configuration, just to check something. I would vote for adding these attributes to default Sylius fixtures file + write PHPUnit test for ProductExampleFactory

**Comment by lchrusciel** [comment]
But huge 👏 for resolving bug itself ;)

File: config/packages/test/fixtures.yaml (hunk 2 above)
Comment (Zales0123) [reply config/packages/test/fixtures.yaml]: I just wanted to be 100% that both `ProductExampleFactory` works fine... On the other hand, to solve the issue indeed the test for only `ProductExampleFactory` would be sufficient 👍

**Comment by pamil** [comment]
The base of this pull-request was changed, you need fetch and reset your local branch 
if you want to add new commits to this pull request. **Reset before you pull, else commits
may become messed-up.**

Unless you added new commits (to this branch) locally that you did not push yet, 
execute `git fetch origin && git reset "product-attribute-fixtures-improvements"` to update your local branch.
 
Feel free to ask for assistance when you get stuck :+1:

**Comment by pamil** [comment]
The base of this pull-request was changed, you need fetch and reset your local branch 
if you want to add new commits to this pull request. **Reset before you pull, else commits
may become messed-up.**

Unless you added new commits (to this branch) locally that you did not push yet, 
execute `git fetch origin && git reset "product-attribute-fixtures-improvements"` to update your local branch.
 
Feel free to ask for assistance when you get stuck :+1:

**Comment by pamil** [comment]
Thank you, Mateusz! :tada:

**Pull request merged**